core/sqf/src/seatrans/tm/hbasetmlib2/javaobjectinterfacetm.cpp

```diff
@@ -478,3 +499,17 @@ void JavaObjectInterfaceTM::appendExceptionMessages(JNIEnv *jenv, jthrowable a_e
     jenv->DeleteLocalRef(a_exception);
 }
 
+short JavaObjectInterfaceTM::getExceptionErrorCode(JNIEnv *jenv, jthrowable a_exception)
+{
+  jshort errCode = JOI_OK;
+  if(a_exception != NULL)
+  {
+    //Only TransactionManagerException class has errorcode defined.
+    if(jenv->IsInstanceOf(a_exception, gTransactionManagerExceptionClass))
+    {
+      errCode =(jshort) jenv->CallShortMethod(a_exception,
+                                              gGetErrorCodeMethodID);
+    }
```

Comment:
Add exception handling code here or at least clear the exception and set the error code to zero